### DISCUSSION TITLE
Can I run EventLoop per cycle instead of as a loop?

### DISCUSSION THREAD

**Discussion by archsolar** [rust-windowing/winit#2437] [Q&A] [answered]
Instead of ``event_loop.run(...)`` 
I'd like:
```
fn tick()
{
     event_loop.run(...)
     //other functions
}
```
Is this possible?

**Comment by archsolar** [comment]
I've been told that this just doesn't work on certain platforms, but EventLoopExt on windows is a possibility.

**Comment by archsolar** [comment]
So guess I'll try to emulate it.

**Comment by archsolar** [comment]
"You are strongly encouraged to use run, unless the use of this is absolutely necessary."

Alright

**Comment by kchibisov** [comment, marked as answer]
You may want to use `run_return` for that and call for it from time to time. You can stop polling events on `RedrawEventsCleared`.

**Comment by ALPHAAAL** [comment]
I am trying to achieve this in wasm.
Start animation -> draw frame 1 - 1000 of the animation -> pause it -> do something on js side -> draw frames 2000 - 3000 (I intentionally skip frames, but it shouldn't matter).

I know that winit doesn't allow creating multiple event loop, so I have to reuse the event loop somehow. However, after finish executing the first 1000 frames and the execution 'goes back' to JS side, I am not able to obtain the event loop object and reuse it for drawing frame 2000 - 3000. May I ask if there is any way to achieve this? 😭 Thank you very much!
